src/Workspaces/Core/Portable/Utilities/EditDistance.cs

```diff
@@ -1,175 +1,506 @@
-                    rowCurrent[0] = rowBefore1[0] + Insert;
+            // The is the minimum number of edits we'd have to make.  i.e. if  'shortString' and 
+            // 'longString' are the same length, then we might not need to make any edits.  However,
+            // if longString has length 10 and shortString has length 7, then we're going to have to
+            // make at least 3 edits.
 
-                    for (int column = 1; column < rowWidth; column++)
-                    {
-                        // Copy = top left neighbor + cost_copy      if current chars are equal
-                        //        infinite                           otherwise
-                        int copyCost = (char.ToLower(oldString[column - 1]) == char.ToLower(newString[row - 1])) ?
-                                            rowBefore1[column - 1] + Copy :
-                                            int.MaxValue;
-
-                        // Insert = top neighbor + cost_insert
-                        int insertCost = rowBefore1[column] + Insert;
-
-                        // Delete = left neighbor + cost_delete
-                        int deleteCost = rowCurrent[column - 1] + Delete;
-
-                        // Replace = top left neighbor + cost_replace
-                        int replaceCost = rowBefore1[column - 1] + Replace;
-
-                        // Twiddle = top left neighbor of the top left neighbor + cost_twiddle   if chars are swapped
-                        //           infinite                                                    otherwise
-                        int twiddleCost = (column > 1 && row > 1 &&
-                                        char.ToLower(oldString[column - 1]) == char.ToLower(newString[row - 2]) &&
-                                        char.ToLower(oldString[column - 2]) == char.ToLower(newString[row - 1])) ?
-                                            rowBefore2[column - 2] + Twiddle :
-                                            int.MaxValue;
-
-                        // Store the smallest of the costs
-                        rowCurrent[column] = Math.Min(Math.Min(Math.Min(Math.Min(copyCost, insertCost), deleteCost), replaceCost), twiddleCost);
-                    }
-                }
+            var minimumEditCount = longLength - shortLength;
+            Debug.Assert(minimumEditCount >= 0);
 
-                // The edit distance is the last element in the current row
-                return rowCurrent[rowWidth - 1];
-            }
-            finally
+            // If our threshold is greater than the number of edits it would take to just produce 'longString'
+            // then just cap the threshold.  In the code below 'threshold' acts as the diagonal
+            // we don't have to look outside of.  By capping the threshold here we make the math
+            // easy below.
+            if (costThreshold > longLength)
```

Comment:
Is this actually necessary? It won't speed anything up and won't bound checks below handle the costThreshold being so large you go outside the matrix?